tests/unsafe-cache.earth

```diff
@@ -0,0 +1,11 @@
+VERSION 0.7
+FROM alpine:3.18
+
+test:
+    RUN --mount=type=cache,target=/cache-test,id=cacheid echo "aaa">/cache-test/test.txt
+    BUILD +check
```

Comment:
There's a race condition here due to `buildAsync`, which will scan the target looking for `BUILD`s that can be pre-executed.

You could do:

```
    RUN --mount=type=cache,target=/cache-test,id=cacheid echo "aaa">/cache-test/test.txt
    WAIT
        BUILD +check
    END
```

to prevent it.

Reply:
here's an example of the race at work:

```
VERSION 0.7
FROM alpine

test:
  RUN echo "in test" && sleep 10 && echo "still in test"
  BUILD +foo

foo:
  RUN echo "in foo"
```

then run

```
alex@surfperch:~/test/build-async$ earthly-v0.7.19 +test
 Init 🚀
————————————————————————————————————————————————————————————————————————————————

           buildkitd | Found buildkit daemon as docker container (earthly-buildkitd)
           buildkitd | Updated image available. Restarting buildkit daemon...
           buildkitd | ...Done

 Build 🔧
————————————————————————————————————————————————————————————————————————————————

              alpine | --> Load metadata alpine linux/amd64
               +base | --> FROM alpine
               +base | [----------] 100% FROM alpine
                +foo | --> RUN echo "in foo"
               +test | --> RUN echo "in test" && sleep 10 && echo "still in test"
               +test | in test
                +foo | in foo
               +test | still in test
              output | --> exporting outputs

 Push Summary ⏫ (disabled)
————————————————————————————————————————————————————————————————————————————————

To enable pushing use earthly --push

 Local Output Summary 🎁
————————————————————————————————————————————————————————————————————————————————
```

Reply:
oh interesting, I thought commands after RUN would be executed after it. Good to know